### DISCUSSION TITLE
Prerendering instead of SSR

### DISCUSSION THREAD

**Discussion by h3** [quasarframework/quasar#9697] [CLI - SSR mode] [answered]
Hello,

I'm trying to implement pre-rendering pattern in a Quasar v2 project.

I have found [a project that has implemented it](https://github.com/stelace/marketplace-demo/blob/dev/quasar.conf.js#L4) with the webpack [prerender-spa-plugin](https://github.com/chrisvfritz/prerender-spa-plugin)

The technique used is discussed in this [feature request thread](https://github.com/quasarframework/quasar/issues/2299#issuecomment-542633771)

My problem is that they use `ssrUpdate`, [which was an internal Quasar API in v1](https://github.com/quasarframework/quasar/issues/8112) that seems to have disappeared in v2.

So this boot pattern does not work anymore:

```
import { Quasar } from 'quasar';
const { ssrUpdate } = Quasar;

export default ({ app }) => {
  ssrUpdate({ app });
};
```

I'm trying to achieve the same in Quasar v2 ..

I'm aware I could use the ssg plugin [discussed here](https://github.com/quasarframework/quasar/issues/2299#issuecomment-812153541), but my project is a SPA and making it SSR compatible is quire a big task at this point.

**Comment by h3** [comment, marked as answer]
Update:

I finally got prerendering working with Quasar v2, it seems the `ssrUpdate` workaround isn't needed anymore.

One caveat is that I did have to use `prerender-spa-plugin-next` instead of `prerender-spa-plugin` due to a Webpack 5 compatibility issue.

**Reply by ontwikkelfabriek** [reply]
@h3 

Ah very good!

**Reply by ivan006** [reply]
@h3 You gonna show us how you did it?

**Comment by capslockoff** [comment]
@h3 I realize It's been a while, but can you share the `quasar.config.js` config relevant to the above? I installed `prerender-spa-plugin-next` but I am running into a weird issue. I can't quite figure out what am I doing wrong.

```
  Error: Prevent writing to file that only differs in casing or query string from
  already written file.
  This will lead to a race-condition and corrupted files on case-insensitive file
  systems.
  d:\code\node\app-web\dist\spa\index.html
  d:\code\node\app-web\dist\spa\index.html
```

**Reply by capslockoff** [reply]
I got this to work with a hack. I added a `postProcess `to rename the output file. Then after the build succeeds, I just manually rename the `index1.html` file to index.html after deleting the existing index.html first. 

```
extendWebpack(cfg, { isServer, isClient }) {
        if (ctx.prod) {
          cfg.plugins.push(
            new PrerenderSPAPlugin({
              indexPath: "index.html",
              staticDir: path.join(__dirname, "dist/spa"),
              routes: pagesToPrerender,
              renderer: require("@prerenderer/renderer-puppeteer"),
              rendererOptions: {
                maxConcurrentRoutes: 1,
                renderAfterTime: 5000,
              },
              postProcess(renderedRoute) {
                if (renderedRoute.route === "/") {
                  renderedRoute.outputPath = "index1.html";
                }
              },
            })
          );
        }
      },
```

**Reply by TobyMosque** [reply]
do u tried this extension?
https://github.com/freddy38510/quasar-app-extension-ssg

**Comment by felek000** [comment]
i want to use `vite-plugin-ssr` any one use this with ssr ? in quasar could share config ?

**Comment by ivan006** [comment]
my solution 

https://github.com/ivan006/2025-08-iframe-vue-cacher-for-seo
